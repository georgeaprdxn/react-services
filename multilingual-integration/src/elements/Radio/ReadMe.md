Radio

```jsx inside Markdown
import mockData from './__test__/Radio.mock.json'
const [isChecked, setIsChecked] = React.useState(false)
const changeHandler = e => setIsChecked(e.target.checked)
;<Radio
	label={mockData.label}
	name={mockData.name}
	id={mockData.id}
	isChecked={isChecked}
	handleChange={e => changeHandler(e)}
/>
```
