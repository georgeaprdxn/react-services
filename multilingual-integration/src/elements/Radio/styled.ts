import styled from 'styled-components'

const StyledRadio = styled.div``
const Input = styled.input``
const Label = styled.label``

export { StyledRadio, Input, Label }
