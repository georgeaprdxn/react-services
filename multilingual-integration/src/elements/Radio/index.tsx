import React, { FunctionComponent } from 'react'

import { StyledRadio, Input, Label } from './styled'

type Props = {
	/**
	 * label to the checkbox
	 */
	label: string
	/**
	 * Name of the checkbox
	 */
	name: string
	/**
	 * unique identifier to the checkbox
	 */
	id: string
	/**
	 * handles change event
	 */
	handleChange?: (e: React.ChangeEvent<HTMLInputElement>) => void
	/**
	 * state of checkbox
	 */
	isChecked: boolean
}

const Radio: FunctionComponent<Props> = ({
	label,
	name,
	id,
	handleChange,
	isChecked,
}) => {
	return (
		<StyledRadio>
			<Label htmlFor={id} className="radio-label">
				<Input
					className="radio-control"
					type="radio"
					id={id}
					name={name}
					onChange={handleChange}
					checked={isChecked}
				/>
				<span className="checkmark" />
				{label}
			</Label>
		</StyledRadio>
	)
}

Radio.defaultProps = {
	handleChange: () => undefined,
}

export default Radio
