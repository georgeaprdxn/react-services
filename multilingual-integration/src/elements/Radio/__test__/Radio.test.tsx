import React from 'react'
import { shallow } from 'enzyme'

import Radio from '../index'
import mockRadioData from './Radio.mock.json'

describe('Radio element tests', () => {
	let radioElem: any = null
	const handleChangeStub = jest.fn()
	beforeEach(() => {
		radioElem = shallow(
			<Radio
				label={mockRadioData.label}
				name={mockRadioData.name}
				id={mockRadioData.id}
				isChecked={mockRadioData.isChecked}
				handleChange={handleChangeStub}
			/>
		)
	})

	it('should render a <input type="radio" />', () => {
		expect(radioElem.find('.radio-control').length).toEqual(1)
	})

	it('should render a <label/>', () => {
		expect(radioElem.find('.radio-label').length).toEqual(1)
	})

	it('should render a <span />', () => {
		expect(radioElem.find('.checkmark').length).toEqual(1)
	})

	it('validate inner text', () => {
		expect(radioElem.find('.radio-label').text()).toEqual(mockRadioData.label)
	})

	it('validates all props', () => {
		expect(radioElem.find('.radio-control').prop('name')).toEqual(
			mockRadioData.name
		)
		expect(radioElem.find('.radio-control').prop('id')).toEqual(
			mockRadioData.id
		)
		expect(radioElem.find('.radio-control').prop('checked')).toEqual(
			mockRadioData.isChecked
		)
	})

	it('checks the DOM events', () => {
		radioElem
			.find(`.radio-control`)
			.simulate('change', { target: { checked: mockRadioData.isChecked } })

		expect(handleChangeStub).toHaveBeenCalledTimes(1)
	})

	it('matches the snaphot', () => {
		expect(radioElem).toMatchSnapshot()
	})
})
