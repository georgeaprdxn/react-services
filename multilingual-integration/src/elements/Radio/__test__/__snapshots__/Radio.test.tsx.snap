// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio element tests matches the snaphot 1`] = `
<styled.div>
  <styled.label
    className="radio-label"
    htmlFor="some-id"
  >
    <styled.input
      checked={true}
      className="radio-control"
      id="some-id"
      name="Some name"
      onChange={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "target": Object {
                  "checked": true,
                },
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      type="radio"
    />
    <span
      className="checkmark"
    />
    Some title
  </styled.label>
</styled.div>
`;
